File: solutions/typescript/01-gg4/code/app/main.ts
import * as fs from 'fs';

const args = process.argv.slice(2);
const command = args[0];

switch (command) {
    case "init":
        fs.mkdirSync(".git", { recursive: true });
        fs.mkdirSync(".git/objects", { recursive: true });
        fs.mkdirSync(".git/refs", { recursive: true });
        fs.writeFileSync(".git/HEAD", "ref: refs/heads/main\n");
        console.log("Initialized git directory");
        break;
    default:
        throw new Error(`Unknown command ${command}`);
}

File: solutions/ocaml/01-gg4/code/src/main.ml
let () =
  let command = Sys.argv.(1) in
  if command = "init" then (
    Unix.mkdir ".git" 0o755;
    Unix.mkdir ".git/objects" 0o755;
    Unix.mkdir ".git/refs" 0o755;
    let oc = open_out ".git/HEAD" in
    output_string oc "ref: refs/heads/main\n";
    close_out oc;
    print_endline "Initialized git directory")
  else
    failwith (Printf.sprintf "Unknown command %s" command)
